.claude/skills/cfn-node-heap-sizer/task-mode-heap-limiter.sh

```
@@ -1,326 +1,326 @@
+}
+EOF
+        echo "📝 Heap configuration logged: $log_file" >&2
+    fi
+}
+
+# Function to validate heap size
+validate_heap_configuration() {
+    local current_heap="${NODE_OPTIONS:-}"
+
+    if echo "$current_heap" | grep -q "max-old-space-size"; then
+        local heap_value=$(echo "$current_heap" | grep -o "max-old-space-size=[0-9]*" | cut -d= -f2)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)